drivers/sdhc/sdhc_esp32.h

```diff
@@ -0,0 +1,155 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SDMMC_FREQ_DEFAULT   20000 /*!< SD/MMC Default speed (limited by clock divider) */
+#define SDMMC_FREQ_HIGHSPEED 40000 /*!< SD High speed (limited by clock divider) */
+#define SDMMC_FREQ_PROBING   400   /*!< SD/MMC probing speed */
+#define SDMMC_FREQ_52M       52000 /*!< MMC 52MHz speed */
+#define SDMMC_FREQ_26M       26000 /*!< MMC 26MHz speed */
+
+/* SD commands remapping */
+#define MMC_WRITE_BLOCK_MULTIPLE SD_WRITE_MULTIPLE_BLOCK
+#define MMC_READ_BLOCK_MULTIPLE  SD_READ_MULTIPLE_BLOCK
+#define MMC_WRITE_DAT_UNTIL_STOP 20
+#define MMC_READ_DAT_UNTIL_STOP  11
+#define MMC_STOP_TRANSMISSION    SD_STOP_TRANSMISSION
+#define MMC_GO_IDLE_STATE        SD_GO_IDLE_STATE
+#define MMC_APP_CMD              SD_APP_CMD
+
+const uint32_t SDMMC_DATA_ERR_MASK = SDMMC_INTMASK_DTO | SDMMC_INTMASK_DCRC | SDMMC_INTMASK_HTO |
+				     SDMMC_INTMASK_SBE | SDMMC_INTMASK_EBE;
+
+const uint32_t SDMMC_DMA_DONE_MASK =
+	SDMMC_IDMAC_INTMASK_RI | SDMMC_IDMAC_INTMASK_TI | SDMMC_IDMAC_INTMASK_NI;
+
+const uint32_t SDMMC_CMD_ERR_MASK = SDMMC_INTMASK_RTO | SDMMC_INTMASK_RCRC | SDMMC_INTMASK_RESP_ERR;
+
+enum sdmmc_req_state {
+	SDMMC_IDLE,
+	SDMMC_SENDING_CMD,
+	SDMMC_SENDING_DATA,
+	SDMMC_BUSY,
+};
+
+/* ESP32 fixed pins for Slot 1 */
+#if CONFIG_SOC_SERIES_ESP32
+#define SDMMC_SLOT1_IOMUX_PIN_NUM_CLK 14
+#define SDMMC_SLOT1_IOMUX_PIN_NUM_CMD 15
+#define SDMMC_SLOT1_IOMUX_PIN_NUM_D0  2
+#define SDMMC_SLOT1_IOMUX_PIN_NUM_D1  4
+#define SDMMC_SLOT1_IOMUX_PIN_NUM_D2  12
+#define SDMMC_SLOT1_IOMUX_PIN_NUM_D3  13
+#endif
+
+/* For debug only */
+static const char *const timingStr[] = {"UNKNOWN", "LEGACY", "HS",    "SDR12", "SDR25", "SDR50",
+					"SDR104",  "DDR50",  "DDR52", "HS200", "HS400"};
+
+struct sdmmc_transfer_state {
+	uint8_t *ptr;
+	size_t size_remaining;
+	size_t next_desc;
+	size_t desc_remaining;
+};
+
+struct sdmmc_event {
+	uint32_t header_DUMMY; /* Reserved for system use (Zephyr message queue) */
+	uint32_t sdmmc_status; /* masked SDMMC interrupt status */
+	uint32_t dma_status;   /* masked DMA interrupt status */
+};
+
+/**
+ * @brief Context of the HAL
+ */
+struct sdmmc_hal_context {
+	sdmmc_dev_t *dev; /* SDMMC SOC layer handle (i.e. register base address) */
+};
+
+/**
+ * Host contexts
+ */
+struct host_ctx {
+	intr_handle_t intr_handle;
+	struct k_msgq *event_queue;
+	struct k_sem io_intr_event;
+	struct sdmmc_hal_context hal;
+};
+
+/**
+ * SD/MMC command information
+ */
+struct sdmmc_command {
+	uint32_t opcode;       /*!< SD or MMC command index */
+	uint32_t arg;          /*!< SD/MMC command argument */
+	uint32_t response[4];  /*!< response buffer */
+	void *data;            /*!< buffer to send or read into */
+	size_t datalen;        /*!< length of data in the buffer */
+	size_t buflen;         /*!< length of the buffer */
+	size_t blklen;         /*!< block length */
+	int flags;             /*!< see below */
+			       /** @cond */
```

Comment:
Why are the `@cond` and `@endcond` present within this file? Ideally, this file will not be parsed by doxygen as it is not a public API.